### DISCUSSION TITLE
Alpine.data not working after hard refresh and then navigating via wire:navigate

### DISCUSSION THREAD

**Discussion by nckrtl** [livewire/livewire#8991] [1. Help] [answered]
I'm having an issue where Alpine.data doesn't seem to work after clicking a link which uses wire:navigate. I get the following error:

<img width="408" alt="Screenshot 2024-11-06 at 20 14 29" src="https://github.com/user-attachments/assets/61172cc7-b335-44ec-81b3-325aa12553cb">

This is the blade of the [page](https://github.com/nckrtl/livewire-test/blob/main/resources/views/livewire/counter1.blade.php) throwing the error when you click on counter one from [this](http://livewire-test.test/counter1) page.

When visiting the counter 1 page directly everything works. When you navigate to the overview and then back it still works. It just doesn't work when you start at the overview page and then go to the counter 1 page.

I tried to listen to both events from alpine and livewire as described here https://github.com/alpinejs/alpine/discussions/3880 but it doesn't seem to work.

You can test it by trying out the linked repo. It's probably something silly but I can't figure it out. Thanks!

**Comment by danie-ramdhani** [comment, marked as answer]
Wrap it inside `@script`

https://livewire.laravel.com/docs/javascript#executing-scripts

**Reply by nckrtl** [reply]
I thought it wasn't working but I needed to get rid of the listener. Thanks!
